darts/timeseries.py

```diff
@@ -3367,20 +3367,22 @@ def resample(
     ) -> Self:
         """
         Build a reindexed ``TimeSeries`` with a given frequency.
-        Provided method is used to fill holes in reindexed TimeSeries, by default 'pad'.
+        Provided method is used to aggregate/fill holes in reindexed TimeSeries, by default 'pad'.
 
         Parameters
         ----------
         freq
             The new time difference between two adjacent entries in the returned TimeSeries.
             Expects a `pandas.DateOffset` or `DateOffset` alias.
         method:
-            Method to fill holes in reindexed TimeSeries (note this does not fill NaNs that already were present):
-
-            'pad': propagate last valid observation forward to next valid
-
-            'backfill': use NEXT valid observation to fill.
+            Method to either aggregate grouped values (for down-sampling) or fill holes (for up-sampling)
```

Comment:
let's chaneg the type from `str` to `Literal`, it will simplify the sanity checks. Since there are a lot of possible values, we could use a constant/alias to store them.

Reply:
As discussed waiting for @dennisbader opinion on this. Typing as Literal is not sufficient for sanity check of `method`. So it would look something like this.

```
ResamplingMethod = Literal["all", ... "var",]

def resample(
        self, freq: Union[str, pd.DateOffset], method: ResamplingMethod = "pad", **kwargs
    ) -> Self:
    ...
    if method in get_args(ResamplingMethod):
    ...
```